### ISSUE TITLE
[Feature request]: Expose "Reconfigure" to Home Assistant or Create Native Auto-Reconfigure for Offline Devices

### ISSUE THREAD

**Issue by sirjmann92** [Koenkk/zigbee2mqtt#21543]
### Is your feature request related to a problem? Please describe

Occasionally devices will fall off the network or go "Offline". There doesn't seem to be a rhyme or reason to which devices, probably caused by temporary interference and lack of response to Ping requests (2/2). A simple Reconfigure in Z2M brings the device back. This doesn't happen very often, and is likely just the nature of a Zigbee network operating on the same frequency as so many other devices in a home (2.4) ,  which makes it difficult to find a proper solution. You can imagine the frustration if one of the devices that goes offline is one that controls lights while you're on the toilet, shower, or a dark room like a basement while you've got your hands full (wife aggro!)

### Describe the solution you'd like

As a workaround, to bring devices back online without having to manually open Z2M, locate any "Offline" devices and select "Reconfigure", it would be nice to be able to create an automation in Home Assistant that listens for devices to go "Offline" and then performs a "Reconfigure" on the device to bring it back.

### Describe alternatives you've considered

An alternative to exposing the Reconfigure as an endpoint to HA would be a native automation or feature within Z2M to continually check for devices that are offline (or at a configurable interval) and perform the Reconfigure. Both solutions would be ideal, although either would satisfy the requirement/request.

### Additional context

I've noticed there are certain endpoints available to Home Assistant now, sensors, switches, etc... like Permit Join, which makes it possible, and very handy, to do things like "Alexa, open Zigbee" with an automation that runs in HA which turns on Z2M's Permit Join with a timer to disable it after a set period (love it!)... Expanding the integration further by exposing additional features to HA would allow for more control and freedom of operation. This feature has the potential to dramatically improve the stability and reliability of a Zigbee network using Zigbee2MQTT (depending on the setup and environmental conditions).

**Comment by DAVIZINH0** [comment]
hello.
this is a good idea. exist any progress or workarround to make this?
maybe in home assistant we can detect when a device is availabe (this is easy)
and then send a mqtt topic to reconfigure this device. is this posible???

**Comment by efenex** [comment]
I have the same need for an automation to be able to trigger reconfigure, as I have aqara switches that will regularly reset and require a reconfigure to function normally again.

**Comment by praun** [comment]
yes, very good idea this. To be honest i was kinda baffled that this wasnt already possible..! 

I have a zigbee device that sometimes gets stuck in the wrong state and then I have to go into z2m and manually reconfigue it, and it instantly pops back into the correct state. 

So i just made a quick google search, to learn how to automate this process, but sadly i just ended up here..

**Comment by boredmthfkr** [comment]
A very good idea. We need this... I am tired of how many times I have to manually trigger the reconfigure for a lightbulb I have that goes offline. Another identical lightbulb, no more than 2 meters away has no issue.

**Comment by GuyFullalove** [comment]
This has my vote :)

**Comment by M4Marc** [comment]
Same Problem anything New about this?

**Comment by Ekylypse** [comment]
You can publish to MQTT topic to do this already. 

Topic: zigbee2mqtt/bridge/request/device/configure

Payload: {"id": "friendly_name"}

Friendly name can also be ieee_address.

This information can be found in the documentation:
https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request

**Comment by M4Marc** [comment]
Hey Anthony,
thank you for your answer.
Now i can try it.Hopfully it will solve my problem.
Very best wishes from Germany
Marc

⁣BlueMail for Android herunterladen ​

**Comment by thebriggsy** [comment]
I could really do with this feature - I'm actually not using Z2MQTT but am using ZHA - I have a sensor which occasionally drops off (every other week or so) and the way I solve it is I press the Reconfigure in the UI. Solves it every time - I'd love to be able to do this via an automation on a regular basis.

**Comment by Ekylypse** [comment]
Well, I haven't read the ZHA docs, but if you'll look at my [previous reply](https://github.com/Koenkk/zigbee2mqtt/issues/21543#issuecomment-2496518686) then you'll see it's already possible to do with an automation with Z2M. I do it with one device that's a pain in the butt.

**Comment by sirjmann92** [comment]
@Ekylypse , this request is to expose a feature in Z2M to HA's API, you can't "do that already". You can, however, use your solution as a workaround to solve the same problem. It's a workaround because you can't use HA's UI to create automations, cards, buttons, etc... without additional effort. Anytime you have to manually use a script or install additional software/plugins, it's a workaround. Some people struggle more than others with scripting, or they just won't use workarounds.

I still appreciate the reply and suggestion. I didn't know it was possible. I also didn't thoroughly read the documentation for possible workarounds or alternative solutions, I just know it's not currently exposed to HA and it would be nice to have.

I'll definitely use this in the short-term for problematic devices, but it would be a bit time-exhaustive to do this for all of my devices.

Thank you!

**Comment by dajomas** [comment]
I just had a couple devices that needed the reconfigure and I would have been very happy if this feature existed.

so here is my vote to add this feature.

**Comment by M4Marc** [comment]
@Ekylypse I've only just been able to devote myself to the topic again and have failed to create an automation. I'm failing at the payload topic, etc. It would be great to get a concrete example. I would like to initiate a reconfiguration of one or better all  offline lamps via an automation.

**Comment by rrozema** [comment]
As @Ekylypse described, even though it's not through a call to Z2M, you can already perform the action from within HA. As an example here's the action you could put into a script or automation to reconfigure a device called "friendly_name". i.e. just replace that string "friendly_name" by your device's name then execute it. I just tested it and it works.
```
action: mqtt.publish
data:
  qos: "0"
  payload: "{\"id\": \"friendly_name\"}"
  topic: zigbee2mqtt/bridge/request/device/configure
  evaluate_payload: false
```

**Comment by n00b001** [comment]
Hi all

I've made this automation for myself, maybe it is of interest to you too

```
alias: "Z2M: Reconfigure Stale or Unavailable Devices"
description: "Once per hour, reconfigure devices if last_seen is 'unavailable' or older than the set threshold."
mode: single
trigger:
  - platform: time_pattern
    hours: "/1"
action:
  # 1. Build the list of devices
  - variables:
      # Threshold in hours for valid timestamps
      hours_offline_threshold: 4
      offline_devices: >-
        {% set threshold = hours_offline_threshold | int %}
        {% set ns = namespace(devices=[]) %}
        
        {# Loop through all sensors ending in _last_seen #}
        {% for s in states.sensor if s.entity_id.endswith('_last_seen') %}
        
          {# Check 1: Is the state explicitly unavailable/unknown? #}
          {% set is_dead_state = s.state in ['unavailable', 'unknown', 'None'] %}
          
          {# Check 2: Calculate time diff (if state is a valid timestamp) #}
          {% set ts = as_timestamp(s.state, 0) %}
          {% set is_stale = (ts > 0) and ((as_timestamp(now()) - ts) > (threshold * 3600)) %}
          
          {# If either condition is true, add to list #}
          {% if is_dead_state or is_stale %}
            {% set fname = s.entity_id | replace('sensor.', '') | replace('_last_seen', '') %}
            {% set ns.devices = ns.devices + [fname] %}
          {% endif %}
          
        {% endfor %}
        {{ ns.devices }}

  # 2. Stop if the list is empty
  - condition: template
    value_template: "{{ offline_devices | length > 0 }}"

  # 3. Loop and Reconfigure
  - repeat:
      for_each: "{{ offline_devices }}"
      sequence:
        - action: mqtt.publish
          data:
            topic: zigbee2mqtt/bridge/request/device/configure
            qos: "0"
            payload: '{"id": "{{ repeat.item }}"}'
            evaluate_payload: false
        # 5 second delay to keep the network stable
        - delay:
            seconds: 5
```
